src/app/credentials.rs

```diff
@@ -1,51 +1,86 @@
 use secret_service::{EncryptionType, Error, SecretService};
-use serde::{Deserialize, Serialize};
-use serde_json::{from_str, to_string};
+use serde::{de::Error as _, ser::Error as _, Deserialize, Deserializer, Serialize, Serializer};
 use std::collections::HashMap;
-use std::str::from_utf8;
+use std::time::{Duration, SystemTime};
 
 static SPOT_ATTR: &str = "spot_credentials";
 
-#[derive(Deserialize, Serialize, Clone, Debug)]
-pub struct Credentials {
-    pub username: String,
-    pub password: String,
-    pub token: String,
-    pub country: String,
-}
-
 fn make_attributes() -> HashMap<&'static str, &'static str> {
     let mut attributes = HashMap::new();
     attributes.insert(SPOT_ATTR, "yes");
     attributes
 }
 
-pub fn try_retrieve_credentials() -> Result<Credentials, Error> {
-    let service = SecretService::new(EncryptionType::Dh)?;
-    let collection = service.get_default_collection()?;
-    let result = collection.search_items(make_attributes())?;
+mod serde_unix_timestamp {
+    use super::*;
 
-    let item = result.get(0).ok_or(Error::NoResult)?.get_secret()?;
-    let raw = from_utf8(&item).unwrap().to_string();
-    let parsed = from_str(&raw).map_err(|_| Error::Parse)?;
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<SystemTime>, D::Error>
```

Comment:
maybe serialization/de serialization should be implemented for SystemTime rather than Option<SystemTime>?